### PR TITLE
add selector functions

### PR DESCRIPTION
**Pull request by smmoosavi**

### CODE REVIEW

File: src/state-selector/create-state-selector.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+export function createStateSelector<V>(
+  options: SelectorOptions<V>,
+): [ReadonlyStateWire<V> | StateWire<V>, ConnectFunction] {
+  const key: string = 'value';
+  const activeWires: {
+    current: Map<ReadonlyStateWire<any>, null | (() => void)>;
+  } = {
+    current: new Map(),
+  };
+  const { get: getOption, set: setOption } = options;
```
Comment (c0m1t) [inline src/state-selector/create-state-selector.ts:45]: Maybe `getOption` and `setOption` names are misleading

File: src/state-selector/create-state-selector.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
+import mitt, { Emitter } from 'mitt';
+import {
+  createReadonlyStateWireGuard,
+  ReadonlyStateWire,
+} from '../state-wire/readonly-state-wire';
+import { createStateWireGuard } from '../state-wire/state-wire';
+import { StateWire } from '../types';
+import { Defined } from '../utils/type-utils';
+
+type DisconnectFunction = () => void;
```
[the remark this answers is not in the document]
Comment (smmoosavi) [reply src/state-selector/create-state-selector.ts:10]: yes. maybe in another PR

File: src/state-selector/create-state-selector.ts (hunk 1 above)
Comment (smmoosavi) [reply src/state-selector/create-state-selector.ts:45]: any suggestions?

Comment (c0m1t) [reply src/state-selector/create-state-selector.ts:45]: `getValue` and `setValue` ?

Comment (smmoosavi) [reply src/state-selector/create-state-selector.ts:45]: these names used at line 98

**Pull request merged**